web/src/page_params.ts

```diff
@@ -24,6 +24,7 @@ export const page_params: {
     }[];
     delivery_email: string;
     development_environment: boolean;
+    furthest_read_time?: number;
```

Comment:
`null`, not `undefined`.

```suggestion
    furthest_read_time: number | null;
```

Reply:
Ah okay. What does it look like in mypy when it's undefined? Or should I always be doing `| null` if it's coming from python?

Reply:
A missing key becomes `undefined` when read. Python `None` translates to JSON `null` (unless there’s some explicit conditional that makes it missing first).